### PR TITLE
create TwoColor.h

### LINKED ISSUE #11: Two-Coloring

### PR DESCRIPTION
**Pull request by azurerazor**

### CODE REVIEW

File: content/graph/TwoColor.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    queue<int> q;
+    for (int i = 0; i < n; i++) {
+        if (col[i] == -1) q.push(i), col[i] = 0;
+        while (q.size()) {
+            int u = q.front(); q.pop();
+            for (auto v : graph[u]) {
+                if (col[v] == -1) {
+                    col[v] = col[u]^1; // sets color to opposite of current
+                    q.push(v);
+                } else if (col[v] == col[u]) return {};
```
Comment (Xylenox) [inline content/graph/TwoColor.h]: nit: this if statement can come first and save the else

**Review by Xylenox — DISMISSED** [review]

**Review by Xylenox — CHANGES_REQUESTED** [review]
could you send a link to the submission that you tested this on?

**Comment by azurerazor** [comment]
@Xylenox 
https://cses.fi/paste/5d2c151ba7202436a24e3a/
https://cses.fi/problemset/task/1668/ (problem statement for above code)

in case that isn't enough:
https://leetcode.com/problems/possible-bipartition/submissions/1403088593

File: content/graph/TwoColor.h (hunk 1 above)
Comment (azurerazor) [reply content/graph/TwoColor.h]: > nit: this if statement can come first and save the else

https://cses.fi/paste/5d2c151ba7202436a24e3a/
https://leetcode.com/problems/possible-bipartition/submissions/1403088593

implemented, although it is now slower

**Review by ssk4988 — APPROVED** [review]
LGTM 👍

**Review by Xylenox — APPROVED** [review]
lgtm

**Pull request merged**